### ISSUE TITLE
Changing a field's sql type from blob to mediumblob isn't recognized by migrate script

### ISSUE THREAD

**Issue by Defcon0** [contao/contao#2750]
**Affected version(s)**

Contao 4.9

**Description**

I need a larger field "tokens" in tl_nc_queue.tokens in the notification center. Therefor I created a DCA file:

```
<?php

$dca = &$GLOBALS['TL_DCA']['tl_nc_queue'];

$dca['fields']['tokens']['sql'] = 'MEDIUMBLOB NULL';
```

In addition I added `notification_center` to the load after array in the `Plugin` class.

Now if I remove the field in the database and run contao:migrate, it works. But if there's already a field with the type blob of the same name, it doesn't alter the field.

Is it a bug? Or is it a quirk in mysql?

Thanks in advance!

Bye Defcon0

**Comment by aschempp** [comment]
This is a known limitation of Doctrine DBAL. I've found this issue https://github.com/doctrine/dbal/issues/2663 but I know we've discussed that before somewhere… not sure if there's fix though.

**Comment by baumannsven** [comment]
Change from a blob to a smaller blob in mysql is not an easy change. You must change it with a migration if the column not empty. Create an new column, then put the data into. After that you remove the old column and rename the new column.

**Comment by Defcon0** [comment]
Thanks for mentioning the issue. I subscribed to it. Maybe another core dev has an opinion if a workaround could be added until the fix in doctrine is done.

**Comment by fritzmg** [comment]
@baumannsven in this case it would be a change to a bigger `MEDIUMBLOB`.

**Comment by baumannsven** [comment]
@fritzmg blob is bigger than medium blob. That is the change. Or i am wrong there?

**Comment by fritzmg** [comment]
`BLOB` » 65536 bytes
`MEDIUMBLOB` » 16777216 bytes

**Comment by Defcon0** [comment]
We need to store more than a blob can take in the tokens field.

**Comment by aschempp** [comment]
> Change from a blob to a smaller blob in mysql is not an easy change. You must change it with a migration if the column not empty. Create an new column, then put the data into. After that you remove the old column and rename the new column.

Same applies to other fields (e.g. varchar).

**Comment by ausi** [comment]
Pull request: https://github.com/doctrine/dbal/pull/4551

**Comment by Defcon0** [comment]
Is this fixed with the above pr?

**Comment by ausi** [comment]
No, https://github.com/doctrine/dbal/pull/4551 was closed in favor of https://github.com/doctrine/dbal/pull/4659

**Comment by Defcon0** [comment]
Would doctrine/dbal#4659 fix the issue here? Since it's a doctrine issue, we can close this ticket anyway, can't we?

**Comment by ausi** [comment]
> Would [doctrine/dbal#4659](https://github.com/doctrine/dbal/pull/4659) fix the issue here?

Yes.

> we can close this ticket anyway, can't we?

👍

**Comment by fritzmg** [comment]
As discussed with @ausi, while debugging an issue similar to [this](https://github.com/contao/contao/issues/3305), we need to upgrade to `doctrine/dbal` `3.x` which has the better/fixed implementation for creating schema diffs and would also solve this issue (see https://github.com/doctrine/dbal/pull/4746).

**Comment by Toflar** [comment]
Just a quick note: This is now fixed with the release of Doctrine DBAL 3.2.0 🎉  I've just tested this locally with the upcoming 4.13 so you'll be all set in February 2022 😊
